src/compiler/checker.ts

```diff
@@ -10377,6 +10381,26 @@ namespace ts {
             return result;
         }
 
+        function isAritySmaller(sourceType: Type, target: FunctionExpression | ArrowFunction) {

```

Comment:
Change this to operate on a single source signature and call it as the last check.